server/models/blob.go

```diff
@@ -61,8 +61,8 @@ func NewBlobForArtifact(artifact *Artifact, contents []byte) *Blob {
 		VersionID:   artifact.VersionID,
 		SpecID:      artifact.SpecID,
 		ArtifactID:  artifact.ArtifactID,
-		Hash:        hashForBytes(contents),
-		SizeInBytes: int32(len(contents)),
+		Hash:        artifact.Hash,
+		SizeInBytes: artifact.SizeInBytes,
```

Comment:
This looks pretty unsafe to me because it relies on the artifact model having these fields computed correctly using the contents that are being provided here. Why are we storing these fields in the Blob _and_ Artifact models anyway?

If Blobs are meant to store contents without knowing the MIME type, then Blobs should not contain the `Hash` and `SizeInBytes` fields that depend on the MIME type. If we are going to store these fields, then we should store the MIME type with it, and these fields should be computed based on the contents and the MIME type.

Reply:
This seems like a good topic for a refactor -- worth considering, but I'd like for this PR to add the capability and pass the tests (and refactor separately)